src/ansys/fluent/core/fluent_connection.py

```diff
@@ -285,6 +361,95 @@ def id(self) -> str:
         """Return the session id."""
         return self._id
 
+    def kill_fluent(self):
+        """
+        Immediately kills the Fluent client,
+        losing unsaved progress and data.
+
+        Examples
+        --------
+
+        >>> import ansys.fluent.core as pyfluent
+        >>> session = pyfluent.launch_fluent()
+        >>> session.kill_fluent()
+
+        Notes
+        -----
+        If the Fluent session is responsive, prefer using :func:`exit()` instead.
+
+        """
+        if self.connection_properties.inside_container:
+            logger.error(
+                f"Cannot execute cleanup script, Fluent running inside container. "
+                f"Use kill_fluent_container() instead."
+            )
+            return
+        if self._remote_instance is not None:
+            logger.error(f"Cannot execute cleanup script, Fluent running remotely.")
+            return
+
+        pwd = self.connection_properties.cortex_pwd
+        pid = self.connection_properties.cortex_pid
+        host = self.connection_properties.cortex_host
+        cleanup_line_start = "Cleanup script file is "
+        if os.name == "nt":
+            cleanup_line_end = ".bat\n"
+            cmd_list = []
+        elif os.name == "posix":
+            cleanup_line_end = ".sh\n"
+            cmd_list = ["bash"]
+        else:
+            logger.error(
+                f"Unrecognized or unsupported operating system, cancelling Fluent cleanup script execution."
+            )
+            return
+        list_dir = os.listdir(Path(pwd))
+        for name in list_dir:
+            if name.startswith("fluent-") and name.endswith(f"{pid}.trn"):
```

Comment:
This assumes Fluent is writing out the automatic transcript, which can be turned off from Preferences.

Another approach is to look for the cleanup script filename pattern in Fluent's working directory.

We will add an API to get the cleanup script name to avoid all these complexities.

Reply:
would the transcript streaming be also a valid data source?

but yeah for all these methods we'll need to have
 - v232 compatible way of doing things
  - v241 cleaner approach, with native apis (as schemeeval is also deprecated)
  -

Reply:
> @mkundu1: This assumes Fluent is writing out the automatic transcript, which can be turned off from Preferences.

That is correct, I was assuming that typical PyFluent use would have the Fluent automatic transcripts enabled, as I believe we currently don't support turning them off through PyFluent (except maybe through tui/scheme-eval?) and auto transcripts are by default enabled on Fluent. 

> @mkundu1: Another approach is to look for the cleanup script filename pattern in Fluent's working directory.

The issue is that the user could and probably often will have multiple cleanup scripts on the same directory for different Fluent sessions, and the cleanup script name has the PID of the host Fluent process, not the Cortex PID, which is what we have access to. The Cortex PID is instead on the Fluent transcript file name. That is why the procedure I set up, to make sure we are killing the correct Fluent client, is: find the transcript for the correct Cortex PID -> find the correct cleanup file inside the transcript and check the hostname matches as well -> execute said cleanup file. 

An alternative would be to parse all cleanup script files in the work dir (as I am not aware of a way to tell which is the correct one, as we can with the transcripts, without going through all scripts in the dir) and then checking every individual PID being killed by the cleanup script, to find the one that matches the Cortex PID, but I believe this is ultimately more prone to failure). I am going to suggest an algorithm that combines both approaches. I am checking if there is a way to get the Fluent host-process PID directly rather than the Cortex PID. Please let me know if you can think of a better way of doing this.

> @mkundu1: We will add an API to get the cleanup script name to avoid all these complexities.

I think that would be the best solution in the long term, but would only be for next year's release of Fluent, right? Or can we do that for Fluent v232 as well? A Fluent settings API command that returns the session's cleanup script name (or specifically the Fluent host PID). That would make things significantly more straightforward. 

> @lagacep-ans: would the transcript streaming be also a valid data source?

Not in the current version. The initial portion of the transcript that has the cleanup script information is not streamed to PyFluent (it is printed out before the server has started and transcript streaming is set up).

Reply:
That is great, tested it and `(cx-client-id)` returns the Fluent host PID that we need. So we can avoid having to find and check Fluent transcript output and simplifies things a bit, thank you for the heads-up